@Types/ISearchCriteria.d.ts

```diff
@@ -0,0 +1 @@
+export type ISearchCriteria = "all" | "vin" | "vrm" | "partialVin" | "trailerId";
```

Comment:
Why do you need a type alias and an enum for the same values?

Reply:
Because you can't use the enum as  a type to limit the string values. 
A "string literal" type lets  you say "only these values are accepted", where as if  you use the enum as the type then you *have* to pass THING.val which is not what actually comes in. Oddly there's no obvious way of to say "I just want the values of this enum as my type please"

Reply:
Meanwhile, I want to compare against the values of the Enum

Reply:
Ah, I see. I was just thinking that we'll have to maintain both files in sync.
Additionally there are no checks at runtime with the string literal type.

Reply:
Nope, but at least it's clear to future Developers what's allowable, and tests can be  representative. I was surprised that a better solution wasn't available but... it wasn't

Reply:
and it will 404 as before if you send "searchCriteria=fish"